### ISSUE TITLE
Blue-green-colourblindness

### ISSUE THREAD

**Issue by Maltragor** [0x90d/videoduplicatefinder#218]
Sorry to open another issue about this, but unfortunately you had neither answered the question about this in #214 (see 2.) nor changed this behaviour in the related commit.
But since the image comparison only takes the red color channel into account, the duplicate detection here (from my point of view) doesn't work as one would expect it spontaneously.
Therefore I ask again for a feedback, whether the behavior is actually so desired.

Current version comparing only red channel:
![image](https://user-images.githubusercontent.com/36803696/134807352-684f6be7-678c-4831-b136-d88569a4d920.png)

Changed to calculated brightness value:
![image](https://user-images.githubusercontent.com/36803696/134807345-0df46084-b78d-4693-8522-706789aa7efb.png)
(Lower threshold set to see all images)

**Comment by 0x90d** [comment]
> whether the behavior is actually so desired.

It has been like that even in VDF 2. I believe it was based on an article but I cannot find it. How does this change perform on a large scan? I wonder if, in normal cirumstances, the other colors matter or not.

**Comment by Maltragor** [comment]
> even in VDF 2

I had also suspected that after seeing that it was not a change as part of the image library migration.
That's why I had only formulated it as a question.

> I believe it was based on an article but I cannot find it. How does this...

I rather thought about the question why all color channels are taken into account for videos, in the form of a merge as grayscale, whereas only the red color channel is taken into account for images. And this still in combination with the fact that the function has grascale in the name and a brightness value is calculated considering all channels, which is then used as toDark criterion. 

The typical weighting factors for the calculation of the grayscale image are probably 0.299 * r + 0.587 * g + 0.114 * b and thus almost correspond to your brightness calculation. I can't explain the difference of 0.0006 for green in your calculation, but it doesn't matter in principle. (Regarding a weighting fits: 0.299 + 0.587 + 0.114 = **1.0**; but since the additional 0.0006*255 = 0.153 related to the rounding and the byte casting can't cause an overflow, at least nothing bad happens here).

> I wonder if, in normal cirumstances, the other colors matter or not.

Here, too, I tend to see it from the other side. The information reduction from RGB to grayscale has quite understandable advantages and one loses "only" the color information. From this point of view, one would have to have a reason / advantage to simply discard two color channels, especially after one has already calculated the brightness value / gray value. Respectively,  if there are any advantages the question would be why not take advantage of it for the videos. (In principle, one would also manage it to extract only one channel with ffmpeg)
From this side my motivation to address this was just that I spontaneously found no explanation for the inconsistency/difference between video and picture "grayscale".

A rather undesirable effect of using only one color channel might be that the usable value/dynamic range depends much more on the scene than when using the brightness, i.e. grayscale. For example, a sunset shifts significantly towards "white", ocean images towards "black". On the other hand, there are of course also scenarios in which an image that is very homogeneous in terms of brightness gains in detail if only considered one color channel, i.e. the value range used is stretched, which may also have positive effects depending on the scenario.
For many images and when choosing a high similarity threshold, I would assume that it does not make a serious difference. (pure assumption, I didn't test it) 
In the end, on the one hand there would be an extreme weighting of only one color channel and thus a potentially stronger dependence / stronger effects depending on the respective image - on the other hand, however, there is the user who wants to compare arbitrary images about whose content one knows nothing, which would rather speak for a neutral weighting behavior. And a brightness value that takes all three colors into account is certainly more neutral than taking only one color into account (especially since this was chosen for the videos). - That would be at least simply my train of thought. But with a study about the effect depending on weighting, image content and threshold I can't serve either ;-)
